packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -539,3 +574,193 @@ mixin _CupertinoSheetRouteTransitionMixin<T> on PageRoute<T> {
     return buildPageTransitions<T>(this, context, animation, secondaryAnimation, child);
   }
 }
+
+class _CupertinoDownGestureDetector<T> extends StatefulWidget {
+  const _CupertinoDownGestureDetector({
+    super.key,
+    required this.enabledCallback,
+    required this.onStartPopGesture,
+    required this.child,
+  });
+
+  final Widget child;
+
+  final ValueGetter<bool> enabledCallback;
+
+  final ValueGetter<_CupertinoDownGestureController<T>> onStartPopGesture;
+
+  @override
+  _CupertinoDownGestureDetectorState<T> createState() => _CupertinoDownGestureDetectorState<T>();
+}
+
+class _CupertinoDownGestureDetectorState<T> extends State<_CupertinoDownGestureDetector<T>> {
+  _CupertinoDownGestureController<T>? _downGestureController;
+
+  late VerticalDragGestureRecognizer _recognizer;
+
+  @override
+  void initState() {
+    super.initState();
+    _recognizer =
+        VerticalDragGestureRecognizer(debugOwner: this)
+          ..onStart = _handleDragStart
+          ..onUpdate = _handleDragUpdate
+          ..onEnd = _handleDragEnd
+          ..onCancel = _handleDragCancel;
+  }
+
+  @override
+  void dispose() {
+    _recognizer.dispose();
+
+    // If this is disposed during a drag, call navigator.didStopUserGesture.
+    if (_downGestureController != null) {
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        if (_downGestureController?.navigator.mounted ?? false) {
+          _downGestureController?.navigator.didStopUserGesture();
+        }
+        _downGestureController = null;
+      });
+    }
+    super.dispose();
+  }
+
+  void _handleDragStart(DragStartDetails details) {
+    assert(mounted);
+    assert(_downGestureController == null);
+    _downGestureController = widget.onStartPopGesture();
+  }
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    assert(mounted);
+    assert(_downGestureController != null);
+    _downGestureController!.dragUpdate(
+      // Devide by size of the sheet. The gap between the top of the sheet and
+      // top of the screen is 0.08.
+      details.primaryDelta! / (context.size!.height - (context.size!.height * 0.08)),
```

Comment:
The unit of 0.08 is screen heights?

Is this something that needs a comment about where got it from?

Reply:
It's the same value used as a part of _kBottomUpTween at the top of the file. Should I put the 0.08 into it's own variable as well you think? Or is that overkill?

Reply:
Oh I see. Could you refer to it as `_kBottomUpTween.end.dy`?